### PR TITLE
Fixes a bug with `rasa data split nlu` 

### PR DESCRIPTION
**Pull request by alopez**
Fix for https://github.com/RasaHQ/rasa/issues/6582 in which `rasa data split nlu` caused the resulting train / test ratio to sometimes differ from the ratio specified by the user or by default.

The splitting algorithm ensures that every intent and response class appears in both the training and the test set. This means that each split must contain at least as many examples as there are classes, which for small datasets can contradict the requested training fraction. When this happens, the command issues a warning to the user that the requested training fraction can't be satisfied.

**Proposed changes**:
- ...

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by sara-tagger** [comment]
Thanks for submitting a pull request 🚀 @b-quachtran will take a look at it as soon as possible ✨

File: tests/shared/nlu/training_data/test_training_data.py [hunk 1]
```diff
@@ -228,34 +228,56 @@ def test_train_test_split(filepaths: List[Text]):
...
-    assert len(trainingdata.number_of_examples_per_response.keys()) == len(
-        trainingdata_test.number_of_examples_per_response.keys()
-    )
-    assert len(trainingdata.number_of_examples_per_response.keys()) == len(
-        trainingdata_train.number_of_examples_per_response.keys()
-    )
+    NUM_TRAIN_EXAMPLES = 46
+    NUM_RESPONSE_EXAMPLES = 4
+
+    assert len(trainingdata.training_examples) == NUM_TRAIN_EXAMPLES
```
Comment (tabergma) [inline tests/shared/nlu/training_data/test_training_data.py:234]: Please rename `trainingdata` to `training_data`

**Review by tabergma — COMMENTED** [review]
Very clean code 💯  Thanks for fixing this!

As this code changes code inside `rasa.shared` someone from the engineering team needs to review this PR. Please request another reviewer. Thanks.

**Comment by wochinge** [comment]
@tabergma Did you review the code inside the `rasa.shared` package as well? Because without looking at the detailed changes, I can say it's okay from a `rasa.shared` perspective as no function names changed

**Comment by wochinge** [comment]
Good catch with `rasa.shared` btw! There was a problem with automated requests for that which we fixed today 👍

**Comment by tabergma** [comment]
@wochinge Yes, the code and the functionality looks good from my side. So, I guess we can approve :)

**Review by wochinge — APPROVED** [review]
